library/multichoicerecyclerview/src/main/java/com/davidecirillo/multichoicerecyclerview/MultiChoiceAdapter.java

```diff
@@ -315,24 +315,12 @@ private void performAll(Action action) {
     }
 
     private void refreshDataSetInternal() {
-        if (mItemList.isEmpty()) {
-            for (int i = 0; i < getItemCount(); i++) {
-                mItemList.put(i, State.INACTIVE);
-            }
-        } else if (mItemList.size() > getItemCount()) { // We have new items, let's add only the new ones as inactive
-            for (int i = 0; i < getItemCount(); i++) {
-                if (mItemList.get(i) != null) {
-                    mItemList.put(i, State.INACTIVE);
-                }
-            }
-        } else {
-            // Keep just the number of item in the getItemCount remove all the others.
-            for (int i = 0; i < mItemList.size(); i++) {
-                if (i + 1 > getItemCount()) {
-                    mItemList.remove(i);
-                }
-            }
+        mItemList.clear();
+
+        for (int i = 0; i < getItemCount(); i++) {
+            mItemList.put(i, State.INACTIVE);
         }
```

Comment:
Yeah, that makes sense, need to handle this differently. Need to completely refresh the data with inactive items and then update the state of the ones that were selected. That way even if the position changes, the same item is selected. Or i could have two different method for this purpose one that keeps the selected items selected. And one that just refresh the data set setting all unselected.